pkg/registry/helm.go

```diff
@@ -0,0 +1,297 @@
+	if localHelmSrcDir == "" {
+		return nil, nil, nil
+	}
+
+	helmManifestPaths, err := getLocalManifestPaths(localHelmSrcDir)
+	if err != nil {
+		return nil, nil, fmt.Errorf("getting helm manifest paths: %w", err)
+	}
+
+	for _, manifest := range helmManifestPaths {
+		helmCRDs, err := parseHelmCRDs(manifest)
+		if err != nil {
+			return nil, nil, fmt.Errorf("parsing helm manifest in '%s': %w", manifest, err)
+		}
+
+		for _, crd := range helmCRDs {
+			var valuesPath string
+
+			if crd.Spec.ValuesContent != "" {
+				valuesPath = filepath.Join(destDir, fmt.Sprintf("values-%s.yaml", crd.Spec.Chart))
+
+				if err = os.WriteFile(valuesPath, []byte(crd.Spec.ValuesContent), fileio.NonExecutablePerms); err != nil {
+					return nil, nil, fmt.Errorf("writing helm values file: %w", err)
+				}
+			}
+
+			if crd.Spec.ChartContent == "" {
+				tempRepo := fmt.Sprintf("repo-%s", crd.Spec.Chart)
+				repository := helmRepositoryName(crd.Spec.Repo, tempRepo, crd.Spec.Chart)
+
+				addCommand := helmAddRepoCommand(crd.Spec.Repo, tempRepo)
+				if addCommand != "" {
+					helmCommands = append(helmCommands, addCommand)
+				}
+
+				templateCommand := helmTemplateCommand(crd, repository, valuesPath, crd.Spec.Chart)
+				pullCommand := helmPullCommand(crd.Spec.Repo, crd.Spec.Chart, crd.Spec.Version, destDir)
+				helmCommands = append(helmCommands, pullCommand, templateCommand)
+				helmChartPaths = append(helmChartPaths, fmt.Sprintf("%s-*.tgz", crd.Spec.Chart))
+			} else {
+				decodedTar, err := base64.StdEncoding.DecodeString(crd.Spec.ChartContent)
+				if err != nil {
+					return nil, nil, fmt.Errorf("decoding base64 chart content: %w", err)
+				}
+				chartTar := filepath.Join(destDir, fmt.Sprintf("%s.tgz", crd.Metadata.Name))
+				err = os.WriteFile(chartTar, decodedTar, fileio.NonExecutablePerms)
+				if err != nil {
+					return nil, nil, fmt.Errorf("writing decoded chart to file: %w", err)
+				}
+
+				templateCommand := helmTemplateCommand(crd, chartTar, valuesPath, crd.Metadata.Name)
+				helmCommands = append(helmCommands, templateCommand)
+				helmChartPaths = append(helmChartPaths, chartTar)
+			}
+		}
+	}
+
+	return helmCommands, helmChartPaths, nil
+}
+
+func helmRepositoryName(repoURL, tempRepo, chart string) string {
+	if strings.HasPrefix(repoURL, "http") {
+		return fmt.Sprintf("%s/%s", tempRepo, chart)
+	}
+
+	return repoURL
+}
+
+func helmAddRepoCommand(repo, tempRepo string) string {
+	if !strings.HasPrefix(repo, "http") {
+		return ""
+	}
+
+	return fmt.Sprintf("helm repo add %s %s", tempRepo, repo)
+}
+
+func helmPullCommand(repository, chart, version string, destDir string) string {
+	repository = helmRepositoryName(repository, fmt.Sprintf("repo-%s", chart), chart)
+
+	pullCommand := fmt.Sprintf("helm pull %s", repository)
+	if version != "" {
+		pullCommand = fmt.Sprintf("helm pull %s --version %s", repository, version)
+	}
+	if destDir != "" {
```

Comment:
Correct me if I'm wrong but we should never end up in a case where `destDir` is empty. Can we enforce it in the `Generate...` function above?

Reply:
You're right, having the option for `""` was useful for testing but no reason to keep it, will update.